### DISCUSSION TITLE
Add menu items that does not refer to a box page

### DISCUSSION THREAD

**Discussion by PubliAlex** [OFFLINE-GmbH/oc-boxes-support#32] [Ideas and feature requests]
Hello,

I need to develop a website that is a mix between boxes pages composed of editable blocks and more complex pages that are developped by me and not editable.

Here is the website menu : 

- Home (boxes)
- About us (boxes)
- Products (not boxes)
- Contact (boxes)

In an ideal scenario, I would like to manage the menu using left panel of boxes, to let the end user change the order of menu items if needed.

That's why I think that it would be great to be able to create a menu item that does not refer to a box page. If you choose "+ create menu item" option, the form on the right would just ask for an URL, or display a list of CMS pages created in the editor to choose from

![image](https://user-images.githubusercontent.com/55833027/197737499-9b67be50-441b-4ab4-8f2c-7783badea9b7.png)

What do you think about that idea ?

Best regards,

**Comment by tobias-kuendig** [comment]
Mixing the pages and menu structure could lead to confusion. But I have to admit, that I wanted to have a proper menu builder integrated in Boxes as well for a long time.

We currently are using the RainLab.Pages menu builder and component to build our menu structure. 

The reason I did not tackle this feature yet is because a new "linker" widget will probably be added to the October CMS core soon. This would be a core component of a menu builder that allows you to link to any model or CMS page of your site.

Let's keep this suggestion int he backlog for a few more weeks to see where the developments in the core go.

**Comment by PubliAlex** [comment]
@tobias-kuendig ok, didn't know that the core team wanted to work on something like this. You are right, if a menu system can be handled by the core, it's a better solution.

**Comment by brightpixels-design** [comment]
@tobias-kuendig I see that OctoberCMS has just released v3.2 which includes a Page Finder form widget - hopefully, is the linker widget you were referring to. https://octobercms.com/blog/post/october-cms-32-new-page-finder-improved-partials-and-more

I am exploring Boxes and its usage, but a menu builder facility is needed for me.  Ideally, one that can handle additional/custom fields so that things like icons, images, colors can be added to the menu item would be great and facilitate a megamenu type capability...

Many thanks for the awesome plugin.

**Comment by PubliAlex** [comment]
Quite agree, I tried to explain to the final customer that pages needs to ba handled with boxes, menu with the menu part of static pages.... nonsense for non october developper. The boxes plugin need a way to handle menu directly from the top left corner of the box plugin itself, with a component to display that menu easily.

![image](https://user-images.githubusercontent.com/55833027/212083729-30858ff2-73ad-4d8c-94dd-6cf55b8df184.png)
